src/rendergraph/opengl/backend/materialshader.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include <QOpenGLShaderProgram>
+#include <QString>
+
+namespace rendergraph::backend {
+class Material;
+class MaterialShader;
+} // namespace rendergraph::backend
+
+class rendergraph::backend::MaterialShader : public QOpenGLShaderProgram {
+  protected:
+    MaterialShader() = default;
+
+  public:
+    int attributeLocation(int attributeIndex) const;
+    int uniformLocation(int uniformIndex) const;
+
+    Material* lastModifiedByMaterial() const;
+    void setLastModifiedByMaterial(Material* pMaterial);
+
+    static QString resource(const char* filename) {
+        return QString(":/shaders/rendergraph/") + QString(filename) + QString(".gl");
```

Comment:
I'm, keeping it simple and stupid :-)

Reply:
But I am okay with using arg. I'll make the change.

Reply:
hardcoding the path here is suboptimal anyways because the library (rendergraph) makes assumptions about the structure of its consumers (mixxx), especially the assumption that it makes use of the Qt resource system (at least thats how I interpret the `:/` prefix). This should at least be changeable at configure time.

Reply:
> But I am okay with using arg. I'll make the change.


don't forget to use `QStringLiteral` on the formatstring ;)

Reply:
> hardcoding the path here is suboptimal anyways because the library (rendergraph) makes assumptions about the structure of its consumers (mixxx), especially the assumption that it makes use of the Qt resource system (at least thats how I interpret the `:/` prefix). This should at least be changeable at configure time.

Let's worry about that when we deploy rendergraph outside of mixxx :-)

Reply:
now using arg and QStringLiteral , please resolve if ok.

Reply:
this seems to have gotten lost somehow